redis/src/cluster_async/mod.rs

```diff
@@ -534,15 +534,18 @@ where
     ) -> RedisResult<ConnectionMap<C>> {
         let initial_nodes: Vec<(String, Option<SocketAddr>)> =
             Self::try_to_expand_initial_nodes(initial_nodes).await;
+        let err = Arc::new(Mutex::new("".to_string()));
```

Comment:
while less critical here, since this isn't a hot loop, there's no need for this allocation.
Instead of converting a `Result` to `Option`, you can just pass the result, and save the last `Err` until you get some `Ok` value.